src/contents/en-gb/doc_xamarin.md

```diff
@@ -0,0 +1,210 @@
+var url = $"safe-auth://{AppId}/{encodedReq}";
+Device.BeginInvokeOnMainThread(() => { Device.OpenUri(new Uri(url)); });
+```
+Make sure you have Authenticator App installed and logged in before running the application.
+
+Once app request is sent, you would notice that Safe Authenticator app is launched and a pop-up dialogue is prompting for access. At this point, you can allow or deny the request, and the Authenticator will send the response back to the application. 
+
+Safe Authenticator application already has the code to process the request and send the response according to user dialogue feedback using the URI scheme, the os will match the URI scheme to our application, and it will launch our application passing the `EncodedAuthResponse` as an argument. We will need to follow some extra steps to extract and use the response from the authenticator. Follow these steps for the respective mobile os.
+
+- Android: We use Intent Filter, so our app can respond to urls. We will add an `IntentFilter` parameter in `MainActivity` attribute. After making this change `MainActivity` should match the following code.
+
+```csharp
+ [Activity(
+        Label = "SafeTodoExample",
+        Icon = "@mipmap/icon",
+        Theme = "@style/MainTheme",
+        MainLauncher = true,
+        ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation),
+        IntentFilter(
+            new[] { Intent.ActionView },
+            Categories = new[] { Intent.CategoryDefault, Intent.CategoryBrowsable },
+            DataScheme = "YOUR_URL_SCHEME"
+        )]
+```
+
+Once this property is set we need to override the `OnNewIntent` action in `MainActivity` to perform next action.
+
+```csharp
+protected override void OnNewIntent(Intent intent)
+{
+    base.OnNewIntent(intent);
+    if (intent?.Data != null)
+    {
+        HandleAppLaunch(intent.Data.ToString());
+    }
+}
+
+private void HandleAppLaunch(string url)
+{
+    System.Diagnostics.Debug.WriteLine($"Launched via: {url}");
+    Device.BeginInvokeOnMainThread(
+        async () =>
+        {
+            try
+            {
+                // Send the url to decode the response and initialize new session (if granted).
+                // Explained in next section
+                
+                System.Diagnostics.Debug.WriteLine("IPC Msg Handling Completed");
+            }
+            catch (Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine($"Error: {ex.Message}");
+            }
+        });
+}
+```
+
+- iOS: We use URL Types Registration feature available in `info.plist` file. We can add it using ui editor or in xml editor. Open `Info.plist` in ui editor and open `Advanced > URL Types > Add URL Type` and add following properties.
+
+```csharp
+Identifier: YOUR_APP_NAME
+URL Schemes: URL_TO_REGISTER
+Role: Viewer
+```
+Once these properties are set then we need to override the `OpenUrl` action in `AppDelegate`.

```

Comment:
Always put a `:` at the end of a sentence which is an introduction to a code block - that way the reader knows the the sentence and code block are related